java-api-core/src/com/kinvey/java/network/NetworkManager.java

```diff
@@ -458,16 +467,20 @@ public String onUnsavedReferenceFound(String collection, GenericJson object) {
 
         if (sourceID != null && !bRealmGeneratedId) {
```

Comment:
Let's add a log statement before this `if` block, so we know when the `ReferenceHelper.processReferences` operation finishes.